lib/block/header.ex

```diff
@@ -284,17 +306,55 @@ defmodule Block.Header do
       ...>  %Block.Header{number: 3_000_000, timestamp: 55, difficulty: 300_000}
       ...> )
       268_734_142
+
+      Test actual Ropsten genesis block
+      iex> Block.Header.get_difficulty(
+      ...>   %Block.Header{number: 0, timestamp: 0},
+      ...>   nil,
+      ...>   0x100000,
+      ...>   0x020000,
+      ...>   0x0800,
+      ...>   0
+      ...> )
+      1_048_576
+
+      # Test actual Ropsten first block
+      iex> Block.Header.get_difficulty(
+      ...>   %Block.Header{number: 1, timestamp: 1_479_642_530},
+      ...>   %Block.Header{number: 0, timestamp: 0, difficulty: 1_048_576},
+      ...>   0x100000,
+      ...>   0x020000,
+      ...>   0x0800,
+      ...>   0
+      ...> )
+      997_888
   """
-  @spec get_difficulty(t, t | nil) :: integer()
-  def get_difficulty(header, parent_header) do
+  @spec get_difficulty(t, t | nil, integer()) :: integer()
+  def get_difficulty(header, parent_header, initial_difficulty \\ @initial_difficulty, minimum_difficulty \\ @minimum_difficulty, difficulty_bound_divisor \\ @difficulty_bound_divisor, homestead_block \\ @homestead_block) do
```

Comment:
Would this work?

    def get_difficulty(header, _,_,_,_,_), when header.number == 0,  do: 0
    def get_difficulty(header, parent_header, initial_difficulty \\ @initial_difficulty, minimum_difficulty \\ @minimum_difficulty, difficulty_bound_divisor \\ @difficulty_bound_divisor, homestead_block \\ @homestead_block) do
     if is_before_homestead?(header, homestead_block) -> do
       ...
     else
       ...
     end